examples/src/bin/gpio_interrupt.rs

```diff
@@ -61,9 +62,9 @@ fn main() -> ! {
     }
 }
 
+#[handler]
```

Comment:
My concern about the `#[handler]` macro is that users used to the rest of the r-e ecosystem will expect it to be called implicitly, but in actual fact, they have to call `set_interrupt_handler` for it to do anything.

Perhaps we can add a [`must_use`](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-must_use-attribute) attribute  to put a warning if you don't set it?

Reply:
Hmm I just tried out must_use locally and don't think it works like that, maybe we can do some type trickery with the macro to create a must_use struct which will print the warning or something.

Reply:
Fair point regarding the user forgetting to call `set_interrupt_handler`. Funny thing is that we get a dead-code warning when the code is just ordinary code but if it's generated via a macro it's suppressed.  (see https://github.com/rust-lang/rust/pull/52467 )

I thought we probably should ditch the idea of the macro but that `extern "C"` somehow makes me feel unconfortable.
Went down the rabbit hole and found a solution. Now commenting out the `set_interrupt_handler` in the example gives me this

```
warning: function `handler` is never used
  --> src\bin\gpio_interrupt.rs:67:4
   |
67 | fn handler() {
   |    ^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

```

I guess that should be warning enough

Reply:
Yeah that should be good enough, it would be nice to customize the lint a bit but I think this is good enough for now. I think most drivers will end up taking the interrupt in the constructor meaning it will be harder to forget to do this, but GPIO is a bit of a special case.